Link existing Trello cards and view list of linked cards

zpawn opened the pull request.

zpawn committed:
Create LinkCards page

zpawn committed:
Merge branch master into feature/sc-74922/trello-link-existing-trello-cards-and-view

zpawn committed:
Add logic to redirect to link page if there are no linked Trello cards for the ticket

zpawn committed:
Merge branch 'master' into feature/sc-74922/trello-link-existing-trello-cards-and-view

zpawn committed:
Perform link of card to ticket using 'associated entity' - this must also add a comment to the Trello card with a hotlink back to the Deskpro ticket

zpawn committed:
List linked Trello cards on home page

zpawn committed:
Allow user to perform search on Trello card titles and IDs on home page

zpawn committed:
Refactoring services

zpawn committed:
Add link to trello card and board

zpawn committed:
fix linter's warnings

zpawn committed:
fix review comments

zpawn committed:
add trello icon

AshleyDawson commented on src/pages/LinkCard.tsx:
```diff
@@ -0,0 +1,45 @@
+
+    return (
+        <>
+            <TwoButtonGroup
```
@zpawn we should hide this for now as the "Create Card" functionality isn't ready yet... don't want customers seeing this until they can use it. Default to link card instead.

AshleyDawson commented on src/pages/LogIn.tsx:
```diff
@@ -21,7 +25,7 @@ const LogIn: FC = () => {
     const [loading, setLoading] = useState<boolean>(false);
     const [error, setError] = useState<string | null>(null);
-    const authUrl = `https://trello.com/1/authorize?expiration=1day&name=MyPersonalToken&scope=read&response_type=token&key=${state?.context?.settings.client_key}&redirect_uri=${callback?.callbackUrl}`
+    const authUrl = `https://trello.com/1/authorize?expiration=1day&name=Deskpro&scope=read&response_type=token&key=${state?.context?.settings.client_key}&redirect_uri=${callback?.callbackUrl}`
```
Change `expiration=1day` to `expiration=never`

AshleyDawson reviewed (changes requested):
Couple of small changes required

zpawn committed:
fix: Change expiration from 1day to never; on LinkPage live onle FindCard component.

zpawn committed:
Change styles for brand link

zpawn committed:
Fix styles for 50% blocks

zpawn committed:
Fix flashes the login screen when loading the app

zpawn committed:
fix linter warnings

AshleyDawson committed:
Style fixes
Added "any" option in board filter
Fixed width issues shen selecting board filter
Moved "linked cards" to top of list for better UX

AshleyDawson committed:
Temp fix for type invariant violations

AshleyDawson committed:
Fixed missing card ID

AshleyDawson committed:
Added cancel button to link cards page

AshleyDawson committed:
Typo fix

AshleyDawson committed:
Silenced type violations
